sched/semaphore/sem_trywait.c

```diff
@@ -69,10 +69,6 @@ int nxsem_trywait(FAR sem_t *sem)
   irqstate_t flags;
   int ret;
 
-  /* This API should not be called from interrupt handlers */
```

Comment:
Why not leave this wrapped in `ifndef CONFIG_DEBUG_MM` ?

Reply:
Since the behavior change happen only when CONFIG_DEBUG_MM is defined, we can simply to:

1. #ifndef CONFIG_DEBUG_MM the below debug assert
2. keep _SEM_TRYWAIT in mm/mm_heap/mm_sem.c with CONFIG_DEBUG_MM guard
3. Remove all other assert

Do you think so? @davids5 .

Reply:
@xiaoxiang781216 Simple is always better. If all code path lead to the one check that would be preferred. If not add them where needed.

Reply:
@davids5 Directly remove is the simply way to sem_trywait(), use CONFIG_DEBUG_MM under `sched` folder is NOT suitable. I strongly suggest keep current code.

Reply:
I see your point. But config.h is global AND you have removed a protection (to facilitate a DEBUG only feature)!

My concern would for a caller of nxsem_trywait (direct) and now the protection is gone. Maybe that does not matter, but if you add 
```
#ifdnef CONFIG_DEBUG_MM
  /* This API should not be called from interrupt handlers */
  DEBUGASSERT(sem != NULL && up_interrupt_context() == false);
 #endif
```
It would help to debug bad calling sequence. Is that only objection having the condition checked CONFIG_DEBUG_MM in 
sched/semaphore/sem_trywait.c?  Have you considered that the cosmetics is minor compared to the debug time that would be spent?

Reply:
(Unresolving this for visibility)

I had the same concern as David's, we will lose the possibility of catching calls from interrupt context to `nxsem_trywait`.

Reply:
Then, there is one question, why user can't call nxsem_trywait from interrupt context ?
There is no harmful, even on SMP mode.
nxsem_trywait only look at the sem count, it will NOT wait all.
If someone use nxsem_trywait, and nxsem_post before the IRQ exit, that's the normal thing I think.

Reply:
ehhh...what is the value of  FAR struct tcb_s *rtcb = this_task(); in the ISR?

Reply:
@davids5 done